docs/architecture.md

```diff
@@ -0,0 +1,121 @@
+
+The GKE Cluster Install step in this repository will build a GKE Private cluster with access to the control plane with the following configuration:
+
+* [Public Endpoint Cluster]() - The cluster can be deployed with public access to the master endpoints therefore eliminating the need for a bastion host. Doing so configures the cluster as follows:
+  * Public endpoint for control plane is enabled
+  * Nodes receive private IP addresses
+  * Restricts access to addresses specified in the authorized networks list
+  * Authorized networks range can contain internal or public IP addresses
+
+Should you want to expose the control plane with a private IP, set `privateEndpoint: true` in your config file. When you choose that setting, the following is configured: 
+
+* [Private Endpoint Cluster]():
+  * Public endpoint for control plane is disabled
+  * Nodes receive private IP addresses
+  * Restricts access to addresses specified in the authorized networks list
+  * Authorized networks range must containe internal IP addresses
+
+The following best practices are also enforced as part of the cluster build process:
+
+* [Shielded VMs and Secure Boot](https://cloud.google.com/kubernetes-engine/docs/how-to/shielded-gke-nodes)
+  * The GKE cluster and bastion host are deployed with Shielded VMs. Doing so provides strong, verifiable node identity and integrity to increase the security of GCE instances.
+  * Additionally, the GKE Shielded nodes are deployed with [Secure Boot](https://cloud.google.com/security/shielded-cloud/shielded-vm#secure-boot). Third-party unsigned kernel modules cannot be loaded when secure boot is enabled.
+
+* [Least Privilege Service Accounts](https://cloud.google.com/kubernetes-engine/docs/how-to/hardening-your-cluster#use_least_privilege_sa)
+  * The build process generates a service account used for running the GKE nodes. This service account operates under the concept of least privilege and only has permissions needed for sending logging data, metrics, and downloading containers from the given GCR project. 
+
+* [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity):
+  * Workload identity is enabled on this cluster and is a way to securely provide access to Google cloud services within your Kubernetes cluster. This allows administrators to bind a Google cloud service account with the roles and/or permissions required to a Kubernetes Service account. An annotation on the service account then references the GCP service account to access the Google cloud services within your cluster.
+
+* [Application Layer Secrets](https://cloud.google.com/kubernetes-engine/docs/how-to/encrypting-secrets#overview):
+  * Application Layer Secrets are used to provide an additional layer of security for sensitive data stored in etcd. The build process creates a [Cloud KMS](https://cloud.google.com/kms/docs) which stores the Key Encrption Key (KEK) used to encrypt data at the application layer. 
```

Comment:
For etcd, link to https://kubernetes.io/docs/concepts/overview/components/#etcd (in case folks are new to K8s)